tf_encrypted/keras/models/sequential.py

```diff
@@ -101,3 +104,113 @@ def layers(self):
     if layers and isinstance(layers[0], InputLayer):
       return layers[1:]
     return layers[:]
+
+  def set_weights(self, weights, sess=None):
+    """ Sets the weights of the model.
+    Arguments:
+      weights: A list of Numpy arrays with shapes and types
+          matching the output of model.get_weights()
+      sess: tfe session"""
+
+    # Updated weights for each layer
+    for layer in self.layers:
+      num_param = len(layer.weights)
+      layer_weights = weights[:num_param]
+      # Define keras weights as private placeholder
+      tfe_weights_pl = [tfe.define_private_placeholder(w.shape)
+                        for w in layer_weights]
+      # Assign new keras weights to existing weights defined by
+      # default when tfe layer was instantiated
+      if not sess:
+        sess = K.get_session()
+      for i, w in enumerate(layer.weights):
+        fd = tfe_weights_pl[i].feed(layer_weights[i])
+        sess.run(tfe.assign(w, tfe_weights_pl[i]), feed_dict=fd)
+
+      weights = weights[num_param:]
+
+  def from_config(self, config):
+    """Instantiates a TFE Keras model from its config.
+
+    Arguments:
+        config: Configuration dictionary.
+            matching the output of model.get_weights()
+        sess: tfe session
+
+    Returns:
+        A TFE Keras model instance
+    """
+    tfe_model = _rebuild_tfe_model(config)
+
+    return tfe_model
+
+def model_from_config(config):
+  """Instantiates a TFE Keras model from its config.
+
+  Arguments:
+      config: Configuration dictionary.
+          matching the output of model.get_weights()
+      sess: tfe session
+
+  Returns:
+      A TFE Keras model instance
+  """
+  return _rebuild_tfe_model(config)
```

Comment:
I would write the logic for `_rebuild_tfe_model` here explicitly, there's no real reason to break it out

Reply:
Good point, let's do that. thanks